core/services/sleeper_task.go

```diff
@@ -61,7 +67,9 @@ func (s *sleeperTask) workerLoop() {
 	for {
 		select {
 		case <-s.waker:
+			s.workerWg.Add(1)
```

Comment:
I'd say this should probably happen before an item is posted on the channel, to avoid a small race.